### ISSUE TITLE
Highlighting is misaligned with text

### ISSUE THREAD

**Issue by jdandrea** [ChrisChinchilla/vale-vscode#52]
In Vale VSCode 0.20.0, I am editing an AsciiDoc file, and I see two problems listed by Vale + VSCode:

- Try to keep sentences short (< 40 words) - from `Microsoft.SentenceLength`
- Consider using 'same' instead of 'identical' - from `Microsoft.ComplexWords`

Yet when I look at the file editor, the yellow squiggle is underlining a completely different part of the file, and not the word 'identical'. And as for the first problem, it's highlighting the end of a sentence with ten words.

Any idea what I might be doing wrong?

**Comment by ChrisChinchilla** [comment]
@jdandrea Yes, I have seen some similar issues, and it seems to be an issue with the actual rules and the `nonword` parameter. Removing that should fix this issue, but may cause others. Once I have confirmation, I will make some PRs to the rule sets to remove that from them.

**Comment by ChrisChinchilla** [comment]
@jdandrea 


I dug into this to see if it's still an issue. Some findings:

This extension doesn't do any range/position translation of its own — diagnostic ranges come straight from vale-ls over LSP.

I also checked whether the nonword theory from my earlier comment actually explains this report: it doesn't. Neither Microsoft.SentenceLength nor Microsoft.ComplexWords has ever used nonword (and ComplexWords was actually merged into Wordiness back in March 2024, before this issue was filed).

 [errata-ai/Microsoft#72](https://github.com/errata-ai/Microsoft/pull/72), fixed last month — but it's in a different rule (FirstPerson), and about 20 other rules still carry nonword: true today.

I tried to reproduce with the current Vale 3.18.0 + Microsoft styles against .adoc content, including inline markup (*bold*, footnote:[...], <<xref>>) around flagged words, and every span landed on exactly the right word.

Could you share (or attach) the actual .adoc file (or a trimmed-down version that still reproduces it), and let me know:

- What OS you're on (Windows/CRLF line endings are a common cause of exactly this "everything past a certain point is offset" symptom)
- The Vale/vale-ls version in use now (a lot may have changed since v0.20.0)
